### PR TITLE
Slightly reduce verbosity when reload_conf is issued

### PR DESCRIPTION
**Pull request by mkavinkumar1**
old state was compared instead of new

### CODE REVIEW

File: freqtrade/worker.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def _worker(self, old_state: Optional[State]) -> State:
         # Log state transition
         if state != old_state:
 
-            if old_state != State.RELOAD_CONFIG:
+            if state != State.RELOAD_CONFIG:
```
Comment (xmatthias) [inline freqtrade/worker.py]: I'm not sure i understand the point of this (what this tries to accomplish).

What this changes is when this is executed (before/after the reload). 
Please explain which messages / in which scenario you think this makes a difference.

**Comment by mkavinkumar1** [comment]
image
The second msg is not necessary

**Review by xmatthias — CHANGES_REQUESTED** [review]
It's incorrect that one of these messages is not needed.

All 3 messages show one particular thing:
1. we received your request to reload the configuration (this is a telegram response - not a state change) - and will not show if you reload via API.
2. we are reloading the configuration (so the bot will now be down for some time)
3. we are now running again.

The mid-state was okish (it changed the wording of the 3rd messages) - but removing this message is not desired, as it hides useful information.

The time between the 2nd and 3rd point can vary greatly depending on the exchange (some exchanges take some time to start up) - so while it might be fast on the configuration you're attempting this - it's not given that that's the case.


image

You're now also hiding the fact of what the new state is. It could be running (default config) - but it could also be stopped (if startup-state is stopped in the configuration).

**Comment by mkavinkumar1** [comment]
> You're now also hiding the fact of what the new state is. It could be running (default config) - but it could also be stopped (if startup-state is stopped in the configuration).

the new state is already ignored.

**Comment by xmatthias** [comment]
> the new state is already ignored.

Which is why i said - the "middle ground" you had was better :)

**Comment by mkavinkumar1** [comment]
> The time between the 2nd and 3rd point can vary greatly depending on the exchange

But the time between 1st and 2nd is instant only. Will the 2nd one fail after 1st?

**Comment by xmatthias** [comment]
> But the time between 1st and 2nd is instant only.

No it's not - it'll depend on the exchange and configuration / setup. 

As mentioned above - the wording can change (which was the first commit) - but the remainder is not desired (so there will be 2 messages - one for "reloading" (start) - and one for "reloaded/running" (end)) - also when reloading via API (which will remove the very first message, which is a response).

The current instance moves further away from that - and is therefore an unwanted change.

**Pull request closed without merging**